pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -232,3 +233,32 @@ def __init__(self, set_obj, *others):
                 raise TypeError(f"Argument of type {o.class_type} cannot be used to build set of type {self._class_type}")
         self._shape = (None,)*self._class_type.rank
         super().__init__(set_obj, *others)
+
+#==============================================================================
+
+class SetIntersection(SetMethod):
+    """
+    Represents a call to the set method .intersection.
+
+    Represents a call to the set method .intersection. This method builds a new set
+    by including all elements which appear in "both" of the iterables
+    (the set object and the arguments).
+
+    Parameters
+    ----------
+    set_obj : TypedAstNode
+        The set object which the method is called from.
+    *others : TypedAstNode
+        The iterables which will be combined (common elements) with this set.
+    """
+    __slots__ = ('_other','_class_type', '_shape')
+    name = 'intersection'
+
+    def __init__(self, set_obj, *others):
+        self._class_type = set_obj.class_type
+        element_type = self._class_type.element_type
+        for o in others:
+            if element_type != o.class_type.element_type:
+                raise TypeError(f"Argument fo type {o.type_class} cannot be used to build set of type {self._class_type}")
+        self._shape = (None,)*self._class_type.rank
```

Comment:
@EmilyBourne, can you please tell me why we are changing to self. rank in the "intersection" and in "union" we have self._class_type.rank?

Reply:
Ah. Feel free to ignore this in that case. I thought `self.rank` would be more readable. It is equivalent to `self._class_type.rank` but are more function calls so it may be slightly slower. I guess that's why I didn't have this change for union

Reply:
Understood.